<launch>

  <arg name="drone_id"/>
  <arg name="drone_id_ns" value="drone$(arg drone_id)"/>
  <arg name="init_x"/>
  <arg name="init_y"/>
  <arg name="init_z"/>
  <!-- num_drones must be at least 2 even if we are planning for a single drone -->
  <arg name="num_drones"/>
  <!-- size of map, change the size inflate x, y, z according to your application -->
  <arg name="map_size_x"/>
  <arg name="map_size_y"/>
  <arg name="map_size_z"/>
  <!-- Input/output topics -->
  <arg name="command_topic"/>
  <arg name="odometry_topic"/>
  <arg name="pose_topic"/>
  <arg name="point_cloud_topic"/>
  <arg name="depth_img_topic"/>

  <!-- Broadcast UAV frames TF -->
  <node pkg="gestelt_bringup" type="uav_tf_broadcaster" 
    name="$(arg drone_id_ns)_tf_broadcaster" output="screen">
    <param name="~drone_id" value="$(arg drone_id)" />
    <param name="~origin_frame" value="$(arg drone_id_ns)_origin" />
    <param name="~base_link_frame" value="$(arg drone_id_ns)_base_link" />
    <param name="~tf_broadcast_freq" value="50.0" />

    <remap from="/mavros/local_position/pose" to="/$(arg drone_id_ns)/mavros/local_position/pose"/>
  </node>

  <include file="$(find gestelt_bringup)/launch/ego_planner.launch">
    <arg name="drone_id"   value="$(arg drone_id)"/>
    <arg name="init_x"     value="$(arg init_x)"/>
    <arg name="init_y"     value="$(arg init_y)"/>
    <arg name="init_z"     value="$(arg init_z)"/>
    <arg name="num_drones" value="$(arg num_drones)"/>
    <arg name="map_size_x" value="$(arg map_size_x)"/>
    <arg name="map_size_y" value="$(arg map_size_y)"/>
    <arg name="map_size_z" value="$(arg map_size_z)"/>
    <arg name="pose_topic" value="$(arg pose_topic)"/>
    <arg name="odometry_topic" value="$(arg odometry_topic)"/>
    <arg name="depth_img_topic" value="$(arg depth_img_topic)"/>
    <arg name="point_cloud_topic" value="$(arg point_cloud_topic)"/>
    <arg name="command_topic" value="$(arg command_topic)" />
  </include>

</launch>
